// Copyright 2015 The gRPC Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";

option java_multiple_files = true;
option java_package = "ru.ifmo.se.protofiles";
option java_outer_classname = "CommunicationProto";
option objc_class_prefix = "HLW";

package communicaiton;

// The greeting service definition.
service Communicator {
  // Sends a greeting
  rpc Poll(EmptyMessage) returns (stream Musician) {}
  rpc Send(Musician) returns (EmptyMessage) {}
}

// The request message containing the user's name.
message Musician {
  string name = 1;
  repeated string generes = 2;
  repeated string tracks = 3;
  string start_time = 4;
  string end_time = 5;
  double x_coord = 6;
  double y_coord = 7;
}

message EmptyMessage {
}
